### PR TITLE
OSDOCS-6811: Added IMDSv2 CLI instructions

### PR DESCRIPTION
**Pull request by EricPonvelle**
Version(s):
`enterprise-4.13+`

Issue:
[OSDOCS-6811](https://issues.redhat.com/browse/OSDOCS-6811)

Link to docs preview:
* [Enabling IMDSv2 in CLI](https://62160--docspreview.netlify.app/openshift-rosa/latest/rosa_cluster_admin/rosa_nodes/rosa-managing-worker-nodes.html#rosa-imds-machine-pools-cli_rosa-managing-worker-nodes) 

QE review:
- [ ] QE has approved this change.

Additional information:
Added instructions for enabling IMDSv2 in the CLI.

### CODE REVIEW

**Comment by sunzhaohua2** [comment]
LGTM

File: modules/rosa-imds-machine-pools-cli.adoc [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+. During the cluster creation process, you see the following line that allows you to specify if you use IMDSv2 only or if you use both IMDSv1 and IMDSv2:
++
+[source,terminal]
+----
+? Configure the use of IMDSv2 for ec2 instances optional/required: required
+----
+
+.Verification
+
+. After your cluster has been created, on your cluster *Overview* tab in {cluster-manager-url}, you see the *Instance Metadata Service (IMDS)* field that notes your IMDS version support.
```
Comment (skopacz1) [inline modules/rosa-imds-machine-pools-cli.adoc:34]: Suggestion to reword this as an imperative for users to follow:

```suggestion
. After your cluster has been created, go to your cluster *Overview* tab in {cluster-manager-url} to see the *Instance Metadata Service (IMDS)* field that notes your IMDS version support.
```

**Review by skopacz1 — COMMENTED** [review]
I left some suggestions for you to consider, otherwise looks good! I am new to reviewing, so please wait until @adellape approves my review before implementing any feedback. Thanks!

File: modules/rosa-imds-machine-pools-cli.adoc [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+.Procedure
+
+. In your terminal, create a ROSA cluster with your specifications. You must provide a value of either `required` or `optional` for the `--ec2-metadata-http-tokens` flag to enable IMDSv2 or a combination of IMDSv1 and IMDSv2.
++
+[source,terminal]
+----
+$ rosa create cluster --cluster <name_of_cluster> --ec2-metadata-http-tokens <required_or_optional>
+----
+. During the cluster creation process, you see the following line that allows you to specify if you use IMDSv2 only or if you use both IMDSv1 and IMDSv2:
```
[the remark this answers is not in the document]
Comment (adellape) [reply modules/rosa-imds-machine-pools-cli.adoc:25]: If this is purely example output from the command with no user input, then I think this suggestion sounds fair (although, also would need a `+` continuation on line 27 in this case).

However, the "that allows you specify" part has me wondering whether this actually does involve user input. So, even if you specify with `--ec2-metadata-http-tokens` in the command whether to use `required` or `optional`, does it still stop you at this stage and allow you confirm or change your choice? Or does this line just fly by w/o being able to be changed.

If there is indeed user input required here (even if just to hit `Enter` to confirm the populated choice from the command flag), then probably keep it as a discrete step but make it a little clearer what's going on there.

File: modules/rosa-imds-machine-pools-cli.adoc (hunk 1 above)
Comment (adellape) [reply modules/rosa-imds-machine-pools-cli.adoc:34]: Or maybe "navigate to" (I see that used a few times elsewhere in the assembly already).

**Review by adellape — COMMENTED** [review]
@skopacz1 Added a few replies, but peer review comments LGTM 👍

File: modules/rosa-imds-machine-pools-cli.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (EricPonvelle) [reply modules/rosa-imds-machine-pools-cli.adoc]: @adellape @skopacz1 - I updated the PR with the suggestions. Can you let me know if I did it correctly/didn't lose the plot?

**Review by EricPonvelle — COMMENTED** [review]
@adellape if you can review the updates, I'd appreciate it.

**Comment by EricPonvelle** [comment]
Re-added the Peer Review Needed label to check over the corrections from the previous review.

File: modules/rosa-imds-machine-pools-cli.adoc [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+.Procedure
+
+. In your terminal, create a ROSA cluster with your specifications by running the following command:
++
+[source,terminal]
+----
+$ rosa create cluster --cluster <name_of_cluster> --ec2-metadata-http-tokens <required_or_optional> <1>
+----
++
+<1> You can provide a value for the `--ec2-metadata-http-tokens` flag, either `required` to enable only IMDSv2 type, or `optional` for a combination of IMDSv1 and IMDSv2. If you do not include this flag, you need to select your IMDS type during the cluster creation prompts.
```
Comment (skopacz1) [inline modules/rosa-imds-machine-pools-cli.adoc:26]: I'd change "you need to" to "you must" per ISG: 

> Use must, not have to or need to, unless must is too restrictive. For example, you do not have to is not the same as you must not.

File: modules/rosa-imds-machine-pools-cli.adoc [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
++
+During the cluster creation process, you see the following line that allows you to specify if you use IMDSv2 only or if you use both IMDSv1 and IMDSv2. If you provided a value when creating your cluster, you need to confirm the selection:
```
Comment (skopacz1) [inline modules/rosa-imds-machine-pools-cli.adoc:28]: In context of the second sentence you added, here is a nit/suggestion:

```suggestion
. Confirm your selection:
```

I'm suggesting making this a step now that I see that it's something the user needs to do, sorry for contradicting my suggestion from the first review. I'm also trimming the context down in the suggestion since I feel like it's implied that the user followed the previous step as instructed, but that's more of a small nit.

File: modules/rosa-imds-machine-pools-cli.adoc [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
++
+[source,terminal]
+----
+? Configure the use of IMDSv2 for ec2 instances optional/required: required
+----
+
+.Verification
+
+. After your cluster has been created, navigate to your cluster *Overview* tab in {cluster-manager-url} to see the *Instance Metadata Service (IMDS)* field that notes your IMDS version support.
```
Comment (skopacz1) [inline modules/rosa-imds-machine-pools-cli.adoc:37]: This still needs to be an unordered list item since this is technically a single step procedure:
```suggestion
* After your cluster has been created, navigate to your cluster *Overview* tab in {cluster-manager-url} to see the *Instance Metadata Service (IMDS)* field that notes your IMDS version support.
```

**Review by skopacz1 — COMMENTED** [review]
Left some comments in light of your changes! @adellape let me know what you think of my new batch of feedback.

File: modules/rosa-imds-machine-pools-cli.adoc (hunk 4 above)
Comment (adellape) [reply modules/rosa-imds-machine-pools-cli.adoc:28]: +1 on making this a step again, and also simplifying the instruction with the assumption they ran the previous step w/ the flag. Possibly a little more context could work, but the even simpler version seems fine too:

```
After running the command, confirm your selection at the following line:
```

Also reminder to remove the `+` continuation on line 27.

**Review by adellape — COMMENTED** [review]
@skopacz1 Updated review comments LGTM 👍
